### ISSUE TITLE
{extension.extensionKey} gets not replaced in templateRootPath

### ISSUE THREAD

**Issue by hahnepeter** [FriendsOfTYPO3/extension_builder#198]
In the generated setup.typoscript, the variable {extension.extensionKey} does not get replaced
for plugin.tx_myext_myplugin.view.templateRootPaths.0
For partialRootPaths.0 and layoutRootPaths.0 it works.

I'm using version 9.10.0

best Regards,
Christian

**Comment by likunstk** [comment]
Confirmed with  master or 9.10  on typo3 9.5.8

**Comment by liayn** [comment]
The related file is https://github.com/FriendsOfTYPO3/extension_builder/blob/master/Resources/Private/CodeTemplates/Extbase/Configuration/TypoScript/setup.tst
I'm pretty sure this is a Fluid problem an not a bug in this extension.

**Comment by vertexvaar** [comment]
With exceptionalErrors set to include warnings i get following error message on save:
```
Extension could not be saved: Could not generate typoscript setup, error: PHP Warning: Invalid argument supplied for foreach() in /app/vendor/typo3fluid/fluid/src/Core/Parser/TemplateProcessor/NamespaceDetectionTemplateProcessor.php line 180
```
It seems either the template in the file `setup.tst` or the parser regex `Patterns::$SPLIT_PATTERN_SHORTHANDSYNTAX` of fluid is broken or we are missing some escaping here, but as far as i can read the regex there is no possibility to escape curly braces.

**Comment by liayn** [comment]
@vertexvaar I assume you use PHP 7.3?
This bug is known in Fluid. PCRE seems to have issues on some Systems. Pretty strange stuff.

**Comment by nicodh** [comment]
Fixed with a3115fc
